.github/workflows/release.yaml

```diff
@@ -0,0 +1,74 @@
+name: release
+on:
+  push:
+    tags:
+      - "2023.*"
+  workflow_dispatch:
+    inputs:
+      snapshot:
+        description: Force a dev version to be generated, implies dry_run.
+        type: boolean
+        required: true
+      dry_run:
+        description: Perform a dry-run release.
+        type: boolean
+        required: true
+
+permissions:
+  # Required to publish a release
+  contents: write
+  # Necessary to push docker images to ghcr.io.
+  packages: write
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    env:
+      # Necessary for Docker manifest
+      DOCKER_CLI_EXPERIMENTAL: "enabled"
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Fetch git tags
+        run: git fetch --tags --force
+
+      - name: Docker Login
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - # Setting up Docker Buildx with docker-container driver is required
+        # at the moment to be able to use a subdirectory with Git context
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: GitHub Tag Name example
+        run: |
+          echo "Tag name from GITHUB_REF_NAME: $GITHUB_REF_NAME"
+          echo "Tag name from github.ref_name: ${{  github.ref_name }}"
+      # - name: Build and push iidoom
+      #   uses: docker/build-push-action@v4
+      #   with:
+      #     context: "{{defaultContext}}:iidoom/build"
+      #     push: true
+      #     tags: "${{github.repository}}:iidoom-${{github.ref_name}}"
+      # https://github.com/marketplace/actions/push-to-ghcr
+      - name: Build ${{github.repository}}/Dockerfile and publish to ${{github.repository_owner}}/infrasnoop:${{github.ref_name}}
+        uses: macbre/push-to-ghcr@master
+        with:
+          context: "."
+          dockerfile: "Dockerfile"
+          image_name: ${{ github.repository_owner }}/infrasnoop # it will be lowercased internally
```

Comment:
This should push to ghcr.io/cncf-infra/infrasnoop:TAG and ghcr.io/cncf-infra/infrasnoop-sideloader:TAG